### PR TITLE
Add HResult, PID and Profiler name to error messages

### PR DESCRIPTION
**Pull request by mattwarren**
See #321

### CODE REVIEW

**Review by WilliamXieMSFT — APPROVED** [review]
:shipit:

**Comment by WilliamXieMSFT** [comment]
>     IfFailRet(InitializeCore(validateCodeSignature));

May be a good idea to also include PID in the IfFailRet macro defined in src/common.lib/macros.h

---
Refers to: src/InstrumentationEngine.Lib/InstrumentationMethod.cpp:54 in 7017d33. [](commit_id = 7017d330cb840a795c3c4beb79ad72bd293800db, deletion_comment = False)

**Review by delmyers — APPROVED** [review]

**Comment by delmyers** [comment]
closes #321

File: src/InstrumentationEngine.Lib/InstrumentationMethod.cpp [hunk 1]
```diff
@@ -74,7 +76,7 @@ HRESULT MicrosoftInstrumentationEngine::CInstrumentationMethod::InitializeCore(
         (m_bstrModuleFolder.Length() == 0) ||
         (m_bstrModule.Length() == 0))
     {
-        CLogging::LogError(_T("CInstrumentationMethod::Initialize - Missing configuration param"));
+        CLogging::LogError(_T("CInstrumentationMethod::Initialize - Missing configuration param, PID: %u"), GetCurrentProcessId());
```
[the remark this answers is not in the document]
Comment (WilliamXieMSFT) [reply src/InstrumentationEngine.Lib/InstrumentationMethod.cpp:79]: The method-specific logger needs to be called from the method itself and prepends the method's class id on those log messages. We could extend that to include PID and other information too.

**Comment by mattwarren** [comment]
@WilliamXieMSFT/@delmyers sorry, I completely missed the notifications from this issue. Thanks for merging the changes in, they'll be very helpful.

Are you able to say when they'll make it into the version of CIE that's deployed in Azure App Services? Or is that something that's out of your hands?

**Comment by delmyers** [comment]
> Are you able to say when they'll make it into the version of CIE that's deployed in Azure App Services? Or is that something that's out of your hands?

@mattwarren We don't have a scheduled release right now. There are at least 2 PRs that need to go in before we release. I expect that it will be within about a month or two, but we generally release as-needed, not on a schedule. Do you have a specific need that would get us to prioritize a release?

**Comment by mattwarren** [comment]
> Do you have a specific need that would get us to prioritize a release?

Just that it's a lot easier for us to diagnose what's going on in Azure App Services with these changes, without them we can't tell which Profiler is doing what.

**Pull request merged**